oxide-auth/src/code_grant/accesstoken.rs

```diff
@@ -563,13 +563,13 @@ pub struct PrimitiveError {
 /// addition this enforces backend specific behaviour for obtaining or handling the access error.
 #[derive(Clone)]
 pub struct ErrorDescription {
-    pub(crate) error: AccessTokenError,
+    pub error: AccessTokenError,
 }
 
 type Result<T> = std::result::Result<T, Error>;
 
 /// Represents an access token, a refresh token and the associated scope for serialization.
-pub struct BearerToken(pub(crate) IssuedToken, pub(crate) String);
+pub struct BearerToken(pub IssuedToken, pub String);
```

Comment:
Making these have all-public attributes fully fixes their representation, which is not desirable. In particular, `String` should not appear in the interface. There could be a constructor from `IssuedToken` and `Scopes` instead. With a warning to only use scopes that come from a grant?

Reply:
So is adding `IssuedToken::convert_bearer_token(self, PreGrant)` okay rather than accepting any `Scope`?